From a2f7f43f461af5f5363035d60ef8880a83a8f0b5 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 13 Jul 2019 17:32:16 -0500
Subject: [PATCH] Sign Pop Event

Fire a cancellable event when a sign tries to pop into an item
---
 src/main/java/net/minecraft/server/BlockFloorSign.java | 10 +++++++++-
 src/main/java/net/minecraft/server/BlockWallSign.java  |  9 ++++++++-
 2 files changed, 17 insertions(+), 2 deletions(-)

diff --git a/src/main/java/net/minecraft/server/BlockFloorSign.java b/src/main/java/net/minecraft/server/BlockFloorSign.java
index 277c881..a87b620 100644
--- a/src/main/java/net/minecraft/server/BlockFloorSign.java
+++ b/src/main/java/net/minecraft/server/BlockFloorSign.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import com.empireminecraft.customevents.SignPopEvent; // EMC
+
 public class BlockFloorSign extends BlockSign {
 
     public static final BlockStateInteger ROTATION = BlockProperties.an;
@@ -20,7 +22,13 @@ public class BlockFloorSign extends BlockSign {
     }
 
     public IBlockData updateState(IBlockData iblockdata, EnumDirection enumdirection, IBlockData iblockdata1, GeneratorAccess generatoraccess, BlockPosition blockposition, BlockPosition blockposition1) {
-        return enumdirection == EnumDirection.DOWN && !this.canPlace(iblockdata, generatoraccess, blockposition) ? Blocks.AIR.getBlockData() : super.updateState(iblockdata, enumdirection, iblockdata1, generatoraccess, blockposition, blockposition1);
+        // EMC Start
+        if (enumdirection == EnumDirection.DOWN && !this.canPlace(iblockdata, generatoraccess, blockposition) && new SignPopEvent(generatoraccess.getMinecraftWorld().getWorld().getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ())).callEvent()) {
+            return Blocks.AIR.getBlockData();
+        } else {
+            return super.updateState(iblockdata, enumdirection, iblockdata1, generatoraccess, blockposition, blockposition1);
+        }
+        // EMC End
     }
 
     public IBlockData a(IBlockData iblockdata, EnumBlockRotation enumblockrotation) {
diff --git a/src/main/java/net/minecraft/server/BlockWallSign.java b/src/main/java/net/minecraft/server/BlockWallSign.java
index 0592a1e..da499ae 100644
--- a/src/main/java/net/minecraft/server/BlockWallSign.java
+++ b/src/main/java/net/minecraft/server/BlockWallSign.java
@@ -1,5 +1,6 @@
 package net.minecraft.server;
 
+import com.empireminecraft.customevents.SignPopEvent; // EMC
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
 import java.util.Map;
@@ -54,7 +55,13 @@ public class BlockWallSign extends BlockSign {
     }
 
     public IBlockData updateState(IBlockData iblockdata, EnumDirection enumdirection, IBlockData iblockdata1, GeneratorAccess generatoraccess, BlockPosition blockposition, BlockPosition blockposition1) {
-        return enumdirection.opposite() == iblockdata.get(BlockWallSign.FACING) && !iblockdata.canPlace(generatoraccess, blockposition) ? Blocks.AIR.getBlockData() : super.updateState(iblockdata, enumdirection, iblockdata1, generatoraccess, blockposition, blockposition1);
+        // EMC Start
+        if (enumdirection.opposite() == iblockdata.get(BlockWallSign.FACING) && !iblockdata.canPlace(generatoraccess, blockposition) && new SignPopEvent(generatoraccess.getMinecraftWorld().getWorld().getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ())).callEvent()) {
+            return Blocks.AIR.getBlockData();
+        } else {
+            return super.updateState(iblockdata, enumdirection, iblockdata1, generatoraccess, blockposition, blockposition1);
+        }
+        // EMC End
     }
 
     public IBlockData a(IBlockData iblockdata, EnumBlockRotation enumblockrotation) {
-- 
2.7.4

